### PR TITLE
Implement geOrchestra's login/logout pattern

### PR DESCRIPTION
**Pull request by jeanpommier**
login = current url + ?login or &login if a query string is already present
logout = /logout

### CODE REVIEW

File: ckanext/georchestra/controller.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+class GeorchestraController(base.BaseController):
+
+    def georchestra_logout(self):
+        '''Override logout with geOrchestra's logout'''
+        return h.redirect_to(u'/logout')
+
+    def georchestra_login(self):
+        '''Override login with geOrchestra's login pattern'''
+        referrer = six.text_type(t.request.referrer, encoding='utf-8')
```
Comment (tonio) [inline ckanext/georchestra/controller.py]: I thought it has only one «r» but looks like both are available in `request`

Screenshot from 2019-04-18 13-12-49

File: ckanext/georchestra/controller.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        url = referrer+u'&login' if '?' in referrer else referrer+u'?login'
```
Comment (tonio) [inline ckanext/georchestra/controller.py]: May be safer with `urlparse`, but good enough.

**Review by tonio — APPROVED** [review]
:+1: (untested, but code looks good)

File: ckanext/georchestra/controller.py (hunk 2 above)
Comment (RemiDesgrange) [reply ckanext/georchestra/controller.py]: +1 for urlparse or string format.

File: ckanext/georchestra/controller.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import ckan.lib.base as base
+import ckan.lib.helpers as h
+import ckan.plugins.toolkit as t
+
+render = base.render
+
+
+class GeorchestraController(base.BaseController):
+
```
Comment (RemiDesgrange) [inline ckanext/georchestra/controller.py:16]: Can you add some docstring please ?

File: ckanext/georchestra/controller.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+import six
+
+import ckan.lib.base as base
+import ckan.lib.helpers as h
+import ckan.plugins.toolkit as t
```
Comment (RemiDesgrange) [inline ckanext/georchestra/controller.py]: Can you import in a pythonic way (`from ckan.libs import helpers`) make more sense than `h` which look like a local variable.

File: ckanext/georchestra/controller.py (hunk 2 above)
Comment (jeanpommier) [reply ckanext/georchestra/controller.py]: OK. Do you know how to set a key-only parameter using urlparse ? I know how to add 'login=true', but just 'login', i don't. Anyway, 'login=true' seems to work.

File: ckanext/georchestra/controller.py (hunk 3 above)
Comment (jeanpommier) [reply ckanext/georchestra/controller.py:16]: Not much to say about this...

File: ckanext/georchestra/controller.py (hunk 4 above)
Comment (jeanpommier) [reply ckanext/georchestra/controller.py]: ok

File: ckanext/georchestra/controller.py (hunk 2 above)
Comment (jeanpommier) [reply ckanext/georchestra/controller.py]: Now using urlparse

**Review by RemiDesgrange — APPROVED** [review]
LGTM :+1:

**Pull request merged**